### ISSUE TITLE
Moving spell range leeway

### ISSUE THREAD

**Issue by SuddenDeath** [LightsHope/server#885]
The Problem: 
Currently there is no "moving range leeway" for ranged spells.
Video I made: https://www.youtube.com/watch?v=5YzXpKaGCe8 (different server but same issue)

How it should work: 
When conditions are met, server should allow +2.66 yards to your ranged spells

Conditions: 
You and your target must both be moving, you both also has to be moving with a "running animation", meaning both of you has to move > 0.7 of normal movement speed(this is where your character switch from walking to running animation).

Examples: 
Use Curse of Exhaustion on target, target is now at 0.7 of normal movement speed, target starts walking.
Use Curse of Exhaustion on target which has a 8% increased movement speed boot enchant, target is now at 0.76 of normal movement speed, target will still be "running".

Proof:
How it's calculated in the 2.4.3 client: http://i.imgur.com/PxKWgdQ.png

Videos of this mechanic:
Retail: https://youtu.be/_DgUcqXwtwk?t=2m
Other server: https://www.youtube.com/watch?v=L8WhIo1l8Aw

How to reproduce: Mimic what is done in the videos.

Basically it's the same mechanic as the melee leeway, except that mechanic should be applied to all spells and not only melee spells.

**Comment by SuddenDeath** [comment]
Also your old commit for the melee leeway is flawed.

Currently on LH all melee spells have 5.33 yard range.
When conditions are met for melee leeway that turns into: 5.33 + 2.5 = 7.83 yards

Which is incorrect, should be 5.0 + 2.66 = 7.66.

Anyway there may be other things that are flawed as well but check my post in this thread #726 for how melee range should be calculated.

**Comment by Cabro** [comment]
Maybe that 5.33 instead of 5.00 is causing the "out of range" bug for yellows while you autoattack?

**Comment by SuddenDeath** [comment]
@Cabro nah that is because of lag/client only updates position every 500ms.

May very well be that auto-attack is handled slightly differently which causes it to be more lenient. I actually remember that happening a lot on retail as well.

**Comment by Cabro** [comment]
@SuddenDeath i hardly doubt it's the 500ms thing. I really have a hard time to accept that. The mobs would reposition after that half a second, yet they don't.
